iceoryx2-pal/configuration/src/lib.rs

```diff
@@ -39,3 +39,17 @@ pub mod settings {
 }
 #[cfg(target_os = "windows")]
 pub use settings::*;
+
+#[cfg(target_os = "macos")]
+pub mod settings {
+    pub const TEMP_DIRECTORY: &[u8] = b"/tmp/";
+    pub const TEST_DIRECTORY: &[u8] = b"/tmp/iceoryx2/tests/";
+    pub const SHARED_MEMORY_DIRECTORY: &[u8] = b"/dev/shm/";
+    pub const PATH_SEPARATOR: u8 = b'/';
+    pub const ROOT: &[u8] = b"/";
+    pub const FILENAME_LENGTH: usize = 255;
+    pub const PATH_LENGTH: usize = 4096;
+    pub const AT_LEAST_TIMING_VARIANCE: f32 = 0.75;
+}
+#[cfg(target_os = "macos")]
+pub use settings::*;
```

Comment:
This does not need to be replicated multiple times with a `cfg` attribute but set only once without the attribute.